File: solutions/csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int timeElapsed) => ExpectedMinutesInOven() - timeElapsed;

    public int PreparationTimeInMinutes(int numberOfLayers) => 2 * numberOfLayers;

    public int ElapsedTimeInMinutes(int numberOfLayers, int cookingTimeElapsed) => cookingTimeElapsed + PreparationTimeInMinutes(numberOfLayers);
}

File: solutions/fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven timeElapsed = expectedMinutesInOven - timeElapsed

let preparationTimeInMinutes layerCount = 2 * layerCount

let elapsedTimeInMinutes layerCount timeElapsed = preparationTimeInMinutes layerCount + timeElapsed